Source/WebKit/UIProcess/API/glib/WebKitURISchemeRequest.cpp

```diff
@@ -204,6 +205,23 @@ SoupMessageHeaders* webkit_uri_scheme_request_get_http_headers(WebKitURISchemeRe
     return request->priv->headers.get();
 }
 
+/**
+ * webkit_uri_scheme_request_get_http_body:
+ * @request: a #WebKitURISchemeRequest
+ *
+ * Get the request body.
+ *
+ * Returns: (transfer full): the body of the @request.
+ *
+ * Since: 2.40
+ */
+GBytes* webkit_uri_scheme_request_get_http_body(WebKitURISchemeRequest* request)
+{
+    g_return_val_if_fail(WEBKIT_IS_URI_SCHEME_REQUEST(request), nullptr);
+
+    return request->priv->task->request().createBodyBytes();
```

Comment:
So we should always use GRefPtr internally. Then here at the outermost layer we can get rid of it:

```suggestion
    return request->priv->task->request().createBodyBytes().leakRef();
```

But it would also be OK to cache a copy like you had been doing with the GIOStream and return `(transfer none)`. I don't have a strong opinion on which way you choose, although I guess caching seems slightly nicer because one benefit of GBytes is that it's refcounted, and that benefit is lost if you create a copy each time. Not that we expect applications will likely be calling this multiple times per request.

Reply:
I changed to transfer full because I saw many APIs do so. I'll switch back to none.